### PR TITLE
Initial state persistence API

### PR DESCRIPTION
**Pull request by connor4312**
Initial implementation, see https://github.com/microsoft/vscode/issues/95361#issuecomment-634319821

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+		}
+	});
+
+	vscode.postMessage({
+		__vscode_notebook_message: true,
+		type: 'initialized'
+	});
+}
+
+export const preloadsScriptStr = (outputNodePadding: number) => `(${webviewPreloads})()`.replace(/__outputNodePadding__/g, `${outputNodePadding}`);
```
Comment (roblourens) [inline src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts:368]: This is much better, thanks!

Comment (rebornix) [reply src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts:407]: nice, love `toString()` ;)

**Review by rebornix — APPROVED** [review]

**Comment by rebornix** [comment]
LGTM, good work!

**Comment by connor4312** [comment]
Thanks ❤️ 

Integration test failures seem existent and unrelated, merging for endgame 🚢

**Pull request merged**